Bottom: 0862c17d0591078130b818a2178a9ab53b817709
Top:    e084c34da0bf71e46d4b5b72a3d5a88299efb0ce
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-03 23:46:22 +0000

Try using cvc_next instead of cvc_prev where possible


---

diff --git a/src/polygon1.c b/src/polygon1.c
index fbf7c8d..3068ebe 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -259,7 +259,7 @@ new_descriptor (VNODE * a, char poly, char side)
     {
       if (side == 'P')
 	{
-	  if (PREV_VERTEX (a)->cvc_prev == (CVCList *) - 1)
+	  if (PREV_VERTEX (a)->cvc_next == (CVCList *) - 1)
 	    PREV_VERTEX (a)->cvc_prev = PREV_VERTEX (a)->cvc_next = NULL;
 	  poly_ExclVertex (PREV_VERTEX (a));
           fprintf (stderr, "REMOVING VERTEX 111111111111111111111111111111111111111111111111111111\n");
@@ -268,7 +268,7 @@ new_descriptor (VNODE * a, char poly, char side)
 	}
       else
 	{
-	  if (NEXT_VERTEX (a)->cvc_prev == (CVCList *) - 1)
+	  if (NEXT_VERTEX (a)->cvc_next == (CVCList *) - 1)
 	    NEXT_VERTEX (a)->cvc_prev = NEXT_VERTEX (a)->cvc_next = NULL;
 	  poly_ExclVertex (NEXT_VERTEX (a));
           fprintf (stderr, "REMOVING VERTEX 222222222222222222222222222222222222222222222222222222\n");
@@ -646,7 +646,7 @@ add_descriptors (PLINE * pl, char poly, CVCList * list)
 
   do
     {
-      if (node->cvc_prev)
+      if (node->cvc_next)
 	{
 	  assert (node->cvc_prev == (CVCList *) - 1
 		  && node->cvc_next == (CVCList *) - 1);
@@ -1770,7 +1770,7 @@ jump (VNODE **curv, DIRECTION *cdir, J_Rule j_rule)
   VNODE *e; /* e is considered an edge */
   DIRECTION newone;
 
-  if (!(*curv)->cvc_prev)	/* not a cross-vertex */
+  if (!(*curv)->cvc_next)	/* not a cross-vertex */
     {
       if (VERTEX_DIRECTION_EDGE (*curv, *cdir)->Flags.mark)
 	return FALSE;
